contracts/BaseGauge.sol

```diff
@@ -58,6 +59,20 @@ abstract contract BaseGauge is IBaseGauge {
         _;
     }
 
+    function setDuration(uint256 newDuration)
+        external
+        onlyOwner
+        updateReward(address(0))
+    {
+        if (block.timestamp < periodFinish) {
+            uint256 remaining = periodFinish - block.timestamp;
+            uint256 leftover = remaining * rewardRate;
+            rewardRate = leftover / newDuration;
+        }
+        duration = newDuration;
```

Comment:
Should we validate newDuration > X? never could be 0 as it'd revert, but is it ok to set it as 1 or we have a minimum value for this?

Reply:
Only 0 is an invalid value and that reverts.